### PR TITLE
CI setup

### PR DESCRIPTION
**Pull request by SevInf**
Multiple different things are happening here:

1. Checked-in pre-compiled engines replaced by the download logic.

   Engine hash is stored in checked-in `.versions/engine` file and is
downloaded in `prepare` script. Pre-compiled engines are still part of
published npm package but they are no longer checked into the
repository. That also removes the need for LFS setup.

2. CI workflow for automated testing is added

   It is pretty much e2e test from prisma/prisma moved into this repository
instead. Build example app and runs simple integration test on both iOS
and Android emulator. It also fixes a bunch of linting failures present
in the repo.

3. CI workflow for automated updates of client, engines, and automated
   publishing of the package.

   Polls npm on cron, if the update found on either `prisma@dev` or
`prisma@latest`, pulls in the new version, does update on the separate
temporary branch and runs aforementioned test on it. If the test is
succsful, branch is merged back into main and new `@prisma/react-native`
package is published with an identical version and npm tag.

4. Bunch of different tools that were added to the dependencies but never actually worked are removed (turborepo, release-it)

Close https://github.com/prisma/team-orm/issues/1106

### CODE REVIEW

File: example/android/app/src/androidTest/java/com/prismaexample/DetoxTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+package com.prismaexample;
```
Comment (SevInf) [inline example/android/app/src/androidTest/java/com/prismaexample/DetoxTest.java:2]: Comes from https://wix.github.io/Detox/docs/introduction/project-setup

File: scripts/check-updates.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      await fs.appendFile(ghOut, `${tag}=${npmVersion}\n`);
+    }
+  } else {
+    console.log('Up to date');
+  }
+}
+
+async function getNpmVersion(tag: NpmTag) {
+  const content = await pRetry(
+    () => $`yarn npm info prisma@${tag} --json -f version`,
```
Comment (Jolg42) [inline scripts/check-updates.ts:31]:
```suggestion
    () => $`npm info prisma@${tag} --json -f version`,
```
No need for yarn right?

Comment (Jolg42) [reply scripts/check-updates.ts:31]: What's `-f` already?

Comment (SevInf) [reply scripts/check-updates.ts:31]: This repo uses yarn exclusively. I briefly tried to convert to `pnpm` and result of it was pretty much that `react-native` and `pnpm` really don't like each other. 

`yarn npm` is not forwarding to npm, it's yarn subcommand for interacting with registry.
`-f` brings back only requested fields from a registry.

**Review by Jolg42 — APPROVED** [review]
Looks good.

I'm not a big fan of having more `yarn` in our codebases but I guess that was not picked by you. It would be nice to get rid of yarn in the future, maybe.

**Pull request merged**